docker/client.py

```diff
@@ -780,7 +780,7 @@ def start(self, container, binds=None, port_bindings=None, lxc_conf=None,
             publish_all_ports=publish_all_ports, links=links, dns=dns,
             privileged=privileged, dns_search=dns_search, cap_add=cap_add,
             cap_drop=cap_drop, volumes_from=volumes_from, devices=devices,
-            network_mode=network_mode, restart_policy=restart_policy,
+            network_mode=network_mode or '', restart_policy=restart_policy,
```

Comment:
It's so that if the user doesn't provide an explicit value, we avoid creating a non-empty config in `start` (see https://github.com/docker/docker-py/blob/jhowardmsft-14530-netmode/docker/utils/utils.py#L431 - not passing `None` will prevent this field from being populated). It's kind of a hack, but the problem is that if a host_config is provided in start it will override the one provided in `create_container`, so we need to prevent that from happening (see 09a5eb8ae79dcba5c232728c1df3644e5bf72c71)

There's an argument for removing config from `start` entirely but I don't know if we're quite there yet.